packages/fuels-accounts/src/provider.rs

```diff
@@ -205,6 +209,10 @@ impl Provider {
         )?;
 
         let (status, receipts) = self.submit_with_feedback(tx.clone()).await?;
+        let receipts = receipts.ok_or(error!(
+            ProviderError,
+            "submitted transaction doesn't have receipts"
```

Comment:
Do recent changes to the fuel-core make it impossible for this to happen unless it is a tx that doesn't generate receipts?
In other words, is it still possible that retrieving receipts will fail (due to sync issues) for a tx that is supposed to have them? If so, we might consider a two step approach to submitting txs and retrieving results on the SDK side.